src/Rules/Variables/CompactVariablesRule.php

```diff
@@ -66,25 +66,23 @@ public function processNode(Node $node, Scope $scope): array
 	}
 
 	/**
-	 * @return array<int, ConstantStringType>
+	 * @return list<ConstantStringType>
 	 */
 	private function findConstantStrings(Type $type): array
 	{
-		if ($type instanceof ConstantStringType) {
-			return [$type];
+		if (count($type->getConstantStrings()) > 0) {
```

Comment:
Introduce a var so you don't need to call `getConstantStrings` twice

Reply:
👍